controllers/profiles/getOtherProfileByUsername.js

```diff
@@ -3,10 +3,26 @@ const {checkMoreOrLess} = require('../../helpers/checkMoreOrLess');
 
 module.exports = async (req, res) => {
   try {
+    // check this route is use auth or not
+    let excludeField = [
+      'human_id',
+      'is_backdoor_user',
+      'encrypted',
+      'created_at',
+      'updated_at',
+      'real_name',
+      'last_active_at',
+      'status',
+      'is_banned'
+    ];
+    if (req?.userId) {
+      excludeField = ['human_id', 'is_backdoor_user', 'encrypted'];
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for excluding fields based on the presence of `req.userId` is clear, but it might be more maintainable to define the common fields to exclude in one array and then add additional fields to exclude if `req.userId` is not present. This way, if you need to change the common fields to exclude, you only have to do it in one place.

```diff
-    let excludeField = [
-      'human_id',
-      'is_backdoor_user',
-      'encrypted',
-      'created_at',
-      'updated_at',
-      'real_name',
-      'last_active_at',
-      'status',
-      'is_banned'
-    ];
-    if (req?.userId) {
-      excludeField = ['human_id', 'is_backdoor_user', 'encrypted'];
-    }
+    const commonExcludeFields = ['human_id', 'is_backdoor_user', 'encrypted'];
+    const additionalExcludeFields = ['created_at', 'updated_at', 'real_name', 'last_active_at', 'status', 'is_banned'];
+    const excludeField = req?.userId ? commonExcludeFields : [...commonExcludeFields, ...additionalExcludeFields];
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->